modules/class-simplechart-api.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Widget data API
+ *
+ * @package Simplechart
+ */
+
+class Simplechart_API {
+
+	public function __construct() {
+		add_action( 'init', array( $this, 'add_rewrite_rule' ) );
+		add_action( 'parse_request', array( $this, 'handle_api_request' ) );
+	}
+
+	public function add_rewrite_rule() {
+		add_rewrite_tag( '%simplechart-api%', '1' );
+		add_rewrite_rule(
+			'^simplechart/api/(\d+)/?',
+			'index.php?p=$matches[1]&post_type=simplechart&simplechart-api=1',
+			'top'
+		);
+	}
+
+	public function handle_api_request( $query ) {
+		if (
+			empty( $query->query_vars['simplechart-api'] ) || empty( $query->query_vars['p'] ) ) {
+			return;
+		}
+		$id = intval( $query->query_vars['p'] );
```

Comment:
probably not a big deal, but `absint()` is a little more thorough than `intval()` (and usually better to use if you are not expecting any negative values).